src/resource.rs

```diff
@@ -40,19 +38,16 @@ pub enum Io {
     Write,
 }
 
-/// Marker traits for types which can be used as a reactor-managed [`Resource`] identifiers.
-pub trait ResourceId: Copy + Eq + Ord + Hash + Send + Debug + Display {}
+/// The resource identifier must be globally unique and non-reusable object. Because of this,
+/// things like [`RawFd`] and socket addresses can't operate like resource identifiers.
+pub type ResourceId = u64;
```

Comment:
Might be a good idea to make this a `struct ResourceId(u64)` to prevent application code from constructing a non-existent resource-id by mistake?

Reply:
Yes, that was a plan. But first I'd like to make sure that everything in heartwood works with this approach - and it isn't. But I can't get your review there since I fail to submit a patch - we just discussed that in chat

Reply:
Ok great :+1: